v2/inc/c_logging/log_context.h

```diff
@@ -4,6 +4,241 @@
+    /* Codes_SRS_LOG_CONTEXT_01_012: [ The name of the struct property shall be the context name specified by using LOG_CONTEXT_NAME (if specified). ]*/ \
+    /* Codes_SRS_LOG_CONTEXT_01_017: [ The name of the struct property shall be the context name specified by using LOG_CONTEXT_NAME (if specified). ]*/ \
+    first_property_value_pair->name = MU_TOSTRING(log_context_name); \
+
+#define EXPAND_SETUP_PROPERTY_PAIR_LOG_CONTEXT_PROPERTY(property_type, property_name, field_value) \
+    /* Codes_SRS_LOG_CONTEXT_01_004: [ LOG_CONTEXT_PROPERTY shall expand to code allocating a property/value pair entry with the type property_type and the name property_name. ]*/ \
+    property_value_pair->value = data_pos; \
+    property_value_pair->name = MU_TOSTRING(property_name); \
+    property_value_pair->type = &property_type##_log_context_property_type; \
+    /* Codes_SRS_LOG_CONTEXT_01_005: [ LOG_CONTEXT_PROPERTY shall expand to code copying the value property_value to be the value of the property/value pair. ]*/ \
+    (void)property_type##_log_context_property_type_init((void*)data_pos, field_value); \
+    data_pos += sizeof(property_type); \
+    property_value_pair++; \
+
+#define SETUP_PROPERTY_PAIR(field_desc) \
+    MU_C2(EXPAND_SETUP_PROPERTY_PAIR_, field_desc)
+
+// COUNT_PROPERTY
+
+#define EXPAND_COUNT_PROPERTY_LOG_CONTEXT_STRING_PROPERTY(property_name, ...) \
+    + 1
+
+#define EXPAND_COUNT_PROPERTY_LOG_CONTEXT_NAME(log_context_name) \
+
+#define EXPAND_COUNT_PROPERTY_LOG_CONTEXT_PROPERTY(property_type, property_name, field_value) \
+    + 1
+
+#define COUNT_PROPERTY(field_desc) \
+    MU_C2(EXPAND_COUNT_PROPERTY_, field_desc)
+
+// COUNT_DATA_BYTES
+
+#define EXPAND_COUNT_DATA_BYTES_LOG_CONTEXT_STRING_PROPERTY(property_name, ...) \
+    + snprintf(NULL, 0, __VA_ARGS__) + 1
+
+#define EXPAND_COUNT_DATA_BYTES_LOG_CONTEXT_NAME(log_context_name) \
+
+#define EXPAND_COUNT_DATA_BYTES_LOG_CONTEXT_PROPERTY(property_type, property_name, field_value) \
+    + sizeof(property_type)
+
+#define COUNT_DATA_BYTES(field_desc) \
+    MU_C2(EXPAND_COUNT_DATA_BYTES_, field_desc)
+
+// Macro that can be used to create a context on the stack
+// We allocate on the stack enough space for a max payload for the context.
+#define LOG_CONTEXT_LOCAL_DEFINE(destination_context, parent_context, ...) \
+    uint8_t MU_C2(values_log_data_, destination_context)[LOG_MAX_STACK_DATA_SIZE]; \
+    LOG_CONTEXT destination_context; \
+    MU_IF(MU_COUNT_ARG(__VA_ARGS__), LOG_CONTEXT_CHECK_VARIABLE_ARGS(__VA_ARGS__),) \
+    LOG_CONTEXT_PROPERTY_VALUE_PAIR MU_C2(property_values_pair_, destination_context)[LOG_MAX_STACK_PROPERTY_VALUE_PAIR_COUNT]; \
+    { \
+        destination_context.values_data = MU_C2(values_log_data_, destination_context); \
+        destination_context.values_data_length = internal_log_context_get_values_data_length_or_zero(parent_context) + 1 MU_IF(MU_COUNT_ARG(__VA_ARGS__), MU_FOR_EACH_1(COUNT_DATA_BYTES, __VA_ARGS__),); \
```

Comment:
>+ 1 [](http://example.com/codeflow?start=117&length=3)

Is this "+1" accounting for some sort of string terminating character?

Reply:
Added an inline comment explaining what it is.